regions/src/main/java/software/amazon/awssdk/regions/providers/InstanceProfileRegionProvider.java

```diff
@@ -28,6 +29,7 @@
  * If {@link SdkSystemSetting#AWS_EC2_METADATA_DISABLED} is set to true, it will not try to load
  * region from EC2 metadata service and will return null.
  */
+@SdkProtectedApi
```

Comment:
Should we just expose these like we do for the credential providers?

Reply:
+1

Reply:
Let's mark them public at GA and leave them be for now?